src/jit/morph.cpp

```diff
@@ -5862,6 +5861,45 @@ GenTreePtr Compiler::fgMorphArrayIndex(GenTreePtr tree)
         elemOffs = offsetof(CORINFO_Array, u1Elems);
     }
 
+#ifndef LEGACY_BACKEND
+    // In minopts, we expand GT_INDEX to GT_IND(GT_INDEX_ADDR) in order to minimize the size of the IR. As minopts
+    // compilation time is roughly proportional to the size of the IR, this helps keep compilation times down.
+    // Furthermore, this representation typically saves on code size in minopts w.r.t. the complete expansion
+    // performed when optimizing, as it does not require LclVar nodes (which are always stack loads/stores in
+    // minopts).
+    if (opts.MinOpts())
```

Comment:
I agree. I intend to do this with a follow-up change, as it requires a number of changes to various bits of the optimizer.